code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/docker/OpendDockerComposeFile.xtend

```diff
@@ -0,0 +1,242 @@
+        )
+
+        service.image = container.name
+        service.container_name = container.name
+        container.endpoints
+
+        container.endpoints.forEach[endpoint |
+            endpoint.addresses.forEach[address |
+               service.ports.add(Util::getPortFromAddress(address) + ':'
+                    + Util::getPortFromAddress(address))
+            ]
+        ]
+
+        this.dockerComposeNetworkParts.forEach[networkPart |
+            service.networks.add(this.getNetworkName(networkPart))
+        ]
+
+        return DockerComposeTemplate::buildDockerComposeParts(container.name, service)
+    }
+
+    /*
+     * Create a POJO [Service] from  a infrastructure node
+     */
+    private def String getServiceFromInfrastructureNode(IntermediateInfrastructureNode node) {
+        val service = new Service
+        val imageTyp = node.defaultValues.findFirst[p |
+            p.technologySpecificProperty.name.toLowerCase == "imagetyp"
+            && p.value.toLowerCase == "springcomponent"
+        ]
+
+        if (imageTyp !== null) {
+            service.build = node.name + File.separator
+        }
+
+        service.image = node.name
+        service.container_name = node.name
+
+        node.endpoints.forEach[endpoint |
+            endpoint.addresses.forEach[address |
+                service.ports.add(Util::getPortFromAddress(address) + ':'
+                    + Util::getPortFromAddress(address))
+            ]
+        ]
+
+        this.dockerComposeNetworkParts.forEach[networkPart |
+            service.networks.add(this.getNetworkName(networkPart))
+        ]
+
+        node.dependsOnNodes.forEach[depentOnNode |
+            service.depends_on.add(depentOnNode.name)
+        ]
+
+        val stringBuilder = new StringBuilder
+        stringBuilder.append(DockerComposeTemplate::buildDockerComposeParts(node.name, service))
+
+        return stringBuilder.toString
+    }
+
+    /*
+     * Get the name for the network in the docker-compose file
+     */
+    private def String getNetworkName(String networkName) {
+        return networkName.lines.toArray.get(0)?.toString.replace(':', "").trim
+    }
+
+    /*
+     * Load a existing docker-compose file from the target folder
+     */
+    private def loadDockerComposePartsFromFile() {
+        val  stringBuilder = new StringBuilder
+        var stream = Files.lines( Paths.get(dockerComposePath), StandardCharsets.UTF_8)
+        stream.forEach[line | stringBuilder.append(line).append("\n")]
+        val representer = new Representer
+        representer.getPropertyUtils().setSkipMissingProperties(true)
+        var yaml = new Yaml(new Constructor(DockerComposeFile),representer)
+        this.buildDockerComposePartFromFile(yaml.loadAs(stringBuilder.toString, DockerComposeFile))
+    }
+
+    /*
+     * Extract the [Service] and [Network] information out of a docker-compose file
+     */
+    private def buildDockerComposePartFromFile(DockerComposeFile file) {
```

Comment:
Hmm, why not call this method `parse`? In fact, it seems to parse a `DockerComposeFile`?